pkg/controller/direct/sql/sqlinstance_controller.go

```diff
@@ -31,14 +31,33 @@ import (
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/config"
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/directbase"
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/registry"
+	kccpredicate "github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/predicate"
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/k8s"
 	"github.com/googleapis/gax-go/v2"
 )
 
 const ctrlName = "sqlinstance-controller"
 
 func init() {
-	registry.RegisterModel(krm.SQLInstanceGVK, newSQLInstanceModel)
+	rolloutGate := &SQLInstanceRolloutGate{}
+	registry.RegisterModelWithRolloutGate(krm.SQLInstanceGVK, newSQLInstanceModel, rolloutGate)
+}
+
+type SQLInstanceRolloutGate struct{}
+
+var _ kccpredicate.RolloutGate = &SQLInstanceRolloutGate{}
+
+func (*SQLInstanceRolloutGate) ShouldReconcile(o *unstructured.Unstructured) bool {
+	u, err := runtime.DefaultUnstructuredConverter.ToUnstructured(o)
```

Comment:
I'm not clear why we need this first ToUnstructured.  It looks like we're converting unstructured -> unstructured?

Reply:
Good point. Looking closer, it seems we can just use the internal `.Object` field as the input to `FromUnstructured`. Got a bit confused on this because `FromUnstructured` needs the `map[string]interface{}` as an input, which isn't exactly the same as `unstructured.Unstructured`.